### PR TITLE
Add Support for logitech G903 Lightspeed wireless mouse.

### PR DESCRIPTION
**Pull request by parkeradam**

### CODE REVIEW

**Comment by perigoso** [comment]
Assuming this is compatible due to similar devices, but do also confirm everything works

File: data/devices/logitech-g903-lightspeed.device [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[Device]
+Name=Logitech G903 Lightspeed
+DeviceMatch=usb:046d:c539
```
[the remark this answers is not in the document]
Comment (Simon1511) [reply data/devices/logitech-g903-lightspeed.device:3]: Just ran `lshid` here with my G903 Lightspeed, running Ubuntu 20.04 with Linux 5.13.

Output is as follows:
```
Device /dev/hidraw6: ID 046d:4067 Logitech G903
Device /dev/hidraw2: ID 046d:c539 Logitech USB Receiver
Device /dev/hidraw3: ID 046d:c539 Logitech USB Receiver
Device /dev/hidraw0: ID 046d:c539 Logitech USB Receiver
Device /dev/hidraw5: ID 1b1c:0a17 Cosair Corsair VOID PRO USB Gaming Headset 
Device /dev/hidraw1: ID 046d:c335 Logitech Gaming Keyboard G910
```

@parkeradam I think this is what @FFY00 was talking about. My G903 (while connected wirelessly) shows up as `46d:4067` for me, while `046d:c539` in fact is only the Receiver.

[the remark this answers is not in the document]
Comment (amfern) [reply data/devices/logitech-g903-lightspeed.device:3]: @Simon1511 Hi,

can you provide the full log of `sudo ratbagd --verbose`

[the remark this answers is not in the document]
Comment (Simon1511) [reply data/devices/logitech-g903-lightspeed.device:3]: > @Simon1511 Hi,
> 
> can you provide the full log of `sudo ratbagd --verbose`

[ratbagd.log](https://github.com/libratbag/libratbag/files/8237651/ratbagd.log)

[the remark this answers is not in the document]
Comment (amfern) [reply data/devices/logitech-g903-lightspeed.device:3]: @Simon1511 you should be able to see the mouse when running `ratbagctl`, can you confirm this?

G903 receiver(c539) is supported by the `hid-logitech-dj.c` driver and your g903 is recognized by libratbag, the product id of your mouse is 4067 which is present in `logitech-g903.device`.

**Comment by amfern** [comment]
@parkeradam can you please try the latest libratbag from github, i suspect you have used an older version libratbag?

If the g903 still isn't recognized, please provide the full log `sudo ratbagd --verbose`

File: data/devices/logitech-g903-lightspeed.device (hunk 1 above)
[the remark this answers is not in the document]
Comment (Simon1511) [reply data/devices/logitech-g903-lightspeed.device:3]: > @Simon1511 you should be able to see the mouse when running `ratbagctl`, can you confirm this?
> 
> G903 receiver(c539) is supported by the `hid-logitech-dj.c` driver and your g903 is recognized by libratbag, the product id of your mouse is 4067 which is present in `logitech-g903.device`.

Yeah that's correct, `ratbagctl` and Piper both detect it as G903.

However. I'm not quite sure how or why, but when I used Piper a few weeks/months ago it wouldn't detect the G903 when connected wirelessly (but `ratbagctl` did detect it). I just checked again with the latest version of Piper on another computer and it appears it works 100% fine now...

So this PR can probably be closed as it works fine when connected both wirelessly and with cable.

[the remark this answers is not in the document]
Comment (amfern) [reply data/devices/logitech-g903-lightspeed.device:3]: Thanks, you testing really helped.
I want to get the log from @parkeradam before closing this PR

**Comment by parkeradam** [comment]
@amfern , apologies for late response, I've moved off Linux since I raised this however it seems to have helped and will hopefully help others in future. Thanks to @Simon1511 for picking up the gauntlet when I was too noob to figure it out. 
Happy for this to be closed @amfern

**Comment by amfern** [comment]
@parkeradam excellent, I will close this PR as g903 seems to work.

**Pull request closed without merging**